src/docfx/lib/HrefUtility.cs

```diff
@@ -24,11 +26,11 @@ public static (string path, string query, string fragment) SplitHref(string href
             var fragmentIndex = href.IndexOf('#');
             if (fragmentIndex >= 0)
             {
-                fragment = href.Substring(fragmentIndex);
+                fragment = href.Substring(fragmentIndex + 1);
```

Comment:
we shouldn't do that, conflicted with fragment definition

Reply:
Yes. I didn't fix it in the first place because of the nasty url concatenation. We can use 'MergeHref' to get rid of it now.